### PR TITLE
Correctly Generate Interface in Generic Class

### LINKED ISSUE #2785: [Bug] Implement interface through static member referred wrong member name in generic classes
I'm using Visual Studio 2015 RC.

**Code example:**

```` C#
class Issue2785<T> : IList<object>
{
    private static List<object> innerList = new List<object>();
}
````

**Note:**

1. The class must be a generic class.
2. The member must be a static field or property.

**Steps:**

1. Put caret on `IList<object>`.
2. Press `Ctrl + .`, choose `Implement interface through "innerList"` menu item.

**What happened:**

Visual Studio generates code using `Issue2785.innerList` to refer this member, causes code failed to compile (missing generic type arguments).

**What should happen:**

Visual Studio uses `innerList` to refer this member, when this member is not static or this class is not generic, Visual Studio will do it correctly.

**Another question:**

When uses "Implement interface explicitly", Visual Studio generates code for members already implemented implicitly, is it a bug or by design?

### PR DESCRIPTION
**Pull request by jmarolf**
Use generic name if class we are generating into is generic.

Fixes #2785

### CODE REVIEW

**Comment by jmarolf** [comment]
@Pilchie @balajikris @dpoeschl @rchande @brettfo @DustinCampbell @basoundr @CyrusNajmabadi - please review

**Comment by Pilchie** [comment]
:+1:

File: src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.CodeAction.cs [hunk 1]
```diff
@@ -407,9 +407,20 @@ private static bool IsUnexpressibleTypeParameter(ITypeParameterSymbol typeParame
 
             private SyntaxNode CreateThroughExpression(SyntaxGenerator factory)
             {
-                var through = ThroughMember.IsStatic
-                    ? factory.IdentifierName(State.ClassOrStructType.Name)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.CodeAction.cs]: i'd prefer you kept the single ternary.  And just updated the true-branch  (to call a helper) that gives you the static "through" name.

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.CodeAction.cs]: Or, do it as:

```c#
var through = !ThroughMember.IsStatic
    ? CreateThis()
    : IsGeneric
        ? ...
        : ...;```

**Pull request merged**